docs/docs/services.md

```diff
@@ -725,3 +725,306 @@ const createPost = (input) => {
+* `prefix`: a global cache key prefix. Defaults to `null`.
+* `fields`: an object that maps the model field names for the `id` and `updatedAt` fields if your database has another name for them. For example: `fields: { id: 'post_id', updatedAt: 'updated_at' }`. Even if only one of your names is different, you need to provide both properties to this option. Defaults to `{ id: 'id', updatedAt: 'updatedAt' }`
+
+### `cache()`
+
+Use this function when you want to cache some data, optionally including a number of seconds before it expires:
+
+```js
+// cache forever
+const post = ({ id }) => {
+  return cache(`posts`, () => {
+    db.post.findMany()
+  })
+}
+
+// cache for 1 hour
+const post = ({ id }) => {
+  return cache(`posts`, () => {
+    db.post.findMany()
+  }, { expires: 3600 })
+}
+```
+
+Note that a key can be a string or an array:
+
+```js
+const post = ({ id }) => {
+  return cache(`posts-${id}-${updatedAt}`, () => {
+    db.post.findMany()
+  })
+}
+
+// or
+
+const post = ({ id }) => {
+  return cache(['posts', id,  updatedAt], () => {
+    db.post.findMany()
+  })
+}
+```
+
+:::info
+
+`cache()` returns a Promise so you'll want to `await` it if you need the data for further processing in your service. If you're only using your service as a GraphQL resolver than you can just return `cache()` directly.
+
+:::
+
+### `cacheFindMany()`
+
+Use this function if you want to cache the results of a `findMany()` call from Prisma, but only until one or more of the records in the set is updated. This is sort of a best of both worlds cache scenario where you can cache as much data as possible, but also expire and re-cache as soon as any piece of it changes, without going through every record manually to see if it's changed: whenever *any* record changes the cache will be discarded.
+
+This function will always execute a `findFirst()` query to get the latest record that's changed, then use its `id` and `updatedAt` timestamp as the cache key for the full query. This means you'll always incur the overhead of a single DB call, but not the bigger `findMany()` unless something has changed. Note you still need to include a cache key prefix:
+
+```js
+const post = ({ id }) => {
+  return cacheFindMany(`users`, db.user)
+}
+```
+
+The above is the simplest usage example. If you need to pass a `where`, or any other object that `findMany()` accepts, include a `conditions` key in an object as the third argument:
+
+```js
+const post = ({ id }) => {
+  return cacheFindMany(`users`, db.user, {
+    conditions: { where: { roles: 'admin' } }
+  })
+}
+```
+
+This is functionally equivalent to the following:
+
+```js
+const latest = db.user.findFirst({
+  where: { roles: 'admin' } },
+  orderBy: { updatedAt: 'desc' },
+  select: { id: true, updatedAt: true
+})
+
+return cache(`posts-${latest.id}-${latest.updatedAt}`, () => {
```

Comment:
I've seen this in some of the examples above. But we recommend in an info block that you use `.getTime()`  on the updatedAt. Since we recommend this I think it is worth putting it in all the examples. The likelihood of people. copying and pasting the examples is probably pretty high and we shouldn't expect them to read the little nuance of the info note we should reiterate it in the examples

```return cache(`posts-${latest.id}-${latest.updatedAt.getTime()}```

Reply:
Agreed